src/database.cpp

```diff
@@ -169,12 +170,23 @@ void Database::deleteOldThumbnails()
         LOG_ERROR() << query.lastError();
 }
 
+void Database::overrideDatabaseDir(QDir *dir)
+{
+    DatabaseDir = dir;
+}
+
 void Database::run()
 {
     connect(&MAIN, SIGNAL(aboutToShutDown()),
             this, SLOT(shutdown()), Qt::DirectConnection);
 
-    QDir dir(QStandardPaths::standardLocations(QStandardPaths::DataLocation).first());
+    QDir dir;
+    if (DatabaseDir != NULL) {
+        dir = DatabaseDir;
+    } else {
+        dir = QStandardPaths::standardLocations(QStandardPaths::DataLocation).first();
```

Comment:
I am thinking we should invert this commit. Instead of storing several globals, setting them, and comparing against them; we should make our DataDir somewhere that is initialized from QStandardPaths, set by command line option, and then read it wherever today we read QStandardPaths::DataLocation.